client/client.go

```diff
@@ -38,8 +45,17 @@ func NewHTTPClient() (*HttpClient, error) {
 		return nil, fmt.Errorf("unable to get valid API key: %w", err)
 	}
 
+	wallet, err := wallets.GetDefaultWallet(store)
+	switch {
+	case errors.Is(err, wallets.ErrNoWalletFound):
+		// No wallet found, continue without wallet.
+	case err != nil:
+		return nil, fmt.Errorf("unable to get default wallet: %w", err)
```

Comment:
although we shouldn't have errors here that arent' bugs, do we want to break any HTTP request (including login and/or signup) if there's any bug in the wallet flow? I think it's fine but I'm not sure if surface it here is better, than surfacing only when it's actually needed

Reply:
That's a good point. Because it is a bug (or someone with a messed up DB) better fail as soon as possible.